users/views.py

```diff
@@ -92,30 +92,15 @@ def index(request: HttpRequest) -> HttpResponse:
     return render(request, "home.html", context={"num_visits": num_visits})
 
 
-class SignUp(APIView):
-    serializer_class = CustomRegisterSerializer
-    renderer_classes = [renderers.TemplateHTMLRenderer]
-    template_name = "signup.html"
-
-    @classmethod
-    def get(cls, request: HttpRequest) -> Response:
-        serializer = CustomRegisterSerializer(context={"request": request})
-        return Response({"serializer": serializer})
-
-    @classmethod
-    def post(cls, request: HttpRequest) -> Union[Response, HttpResponseRedirectBase]:
-        serializer = CustomRegisterSerializer(data=request.data)
-        if not serializer.is_valid():
-            return Response(
-                {
-                    "serializer": serializer,
-                    "errors": serializer.errors,
-                    "non_field_errors": serializer.errors.get("non_field_errors"),
-                }
-            )
-        else:
-            serializer.save(request)
-            return redirect("login")
+def signup(request: HttpRequest) -> HttpResponse:
```

Comment:
Use subclass of Django `LoginView`.

Reply:
I didn't found LoginView subclass of Django to SignUp but I found FormView and wrote it in generic way as I think you wanted.

Reply:
It is here https://github.com/django/django/blob/master/django/contrib/auth/views.py#L38 . You should reuse as much of it as possible, basically I think that it should be enough to:
- override this view class.
- set custom template in overridden class.
- override `get_context_data` if any extra context data is needed.
- set custom form inheriting from `AuthenticationForm` if any modification is needed.

Sorry for late description on this but I wasn't sure how much we want to achieve with the custom sign up view, and now I have checked related issue that basically we want to get rid of `APIView`. So I think we should use `LoginView` as much as possible and override only what is needed.